app_allocator/classes/queue.py

```diff
@@ -0,0 +1,70 @@
+from app_allocator.classes.assignments import has_been_assigned
+
+
+class Queue(object):
+    def __init__(self, constraints={}, count=1):
+        self.constraints = constraints.copy()
+        self.count = count
+        self.counts = {}
+        self.items = []
+        self.assigned = {}
+
+    def __str__(self):
+        return "Queue(%s)" % self.constraints
+
+    def add_constraint(self, constraint):
+        self.constraints.update(constraint)
+
+    def add_if_matches(self, entity):
+        for feature, option in self.constraints.items():
+            if entity.properties.get(feature) != option:
+                return
+        self.items.append(entity)
+
+    def next_item(self, judge):
+        if not self.matches(judge):
+            return None
+        index = 0
+        count = len(self.items)
+        while index < count:
```

Comment:
It would be more idiomatic to iterate over the objects in the list. 
```
for startup in self.items:
  if not self.has_been_assigned(judge, startup):
    ...
```

Reply:
Done.